### DISCUSSION TITLE
How to use Xaml Islands in a .NET WPF App?

### DISCUSSION THREAD

**Discussion by AchimStuy** [microsoft/WindowsAppSDK#3940] [Q&A] [answered]
With the [release of Xaml Islands for Windows App SDK](https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/stable-channel#xaml-islands-no-longer-experimental) - how to use WinUI 3 Controls in a WPF app?

I am stuck with a `COMException`: _The group or resource is not in the correct state to perform the requested operation._

It is documented, how to use Windows App SDK features like ResourceManager (https://learn.microsoft.com/en-us/windows/apps/windows-app-sdk/wpf-plus-winappsdk), but all search for xaml islands leads to the old Community Toolkit documentation.

**Comment by castorix** [comment, marked as answer]
I had posted a test sample ([WPF_XAML_Islands_WinUI3](https://github.com/castorix/WPF_XAML_Islands_WinUI3))

**Comment by DarranRowe** [comment]
As a reference, this is the [C++ sample](https://github.com/microsoft/WindowsAppSDK-Samples/tree/main/Samples/Islands).
This includes everything you need to emulate, including the call to ContentPreTranslateMessage and the stuff for navigation into and out of the island.

**Comment by myth384** [comment]
[Microsoft.Toolkit.Win32](https://github.com/CommunityToolkit/Microsoft.Toolkit.Win32) has a XAML Control [WindowsXamlHost](https://learn.microsoft.com/en-us/dotnet/api/microsoft.toolkit.forms.ui.xamlhost.windowsxamlhost?view=win-comm-toolkit-dotnet-6.1) to host a UWP Control inside WPF or Winforms projects. It acts as a convenience wrapper around `DesktopWindowXamlSource`.  This feature was removed in Windows Community Toolkit 8, reasoning that Windows App SDK 1.4 introduced XAML Islands... as you may have guessed in the form of `DesktopWindowXamlSource`...
I created an [issue](https://github.com/CommunityToolkit/Windows/issues/338) on Github of Windows Community Toolkit to reintroduce the wrappers to work with WinUI 3.
